packages/block-library/src/image/save.js

```diff
@@ -46,10 +46,17 @@ export default function save( { attributes } ) {
 		[ `wp-image-${ id }` ]: !! id,
 	} );
 
+	const describedById = `wp-image-caption-${ id }`;
```

Comment:
From my testing, image `id` can be `undefined` when setting an Image block's media via "Insert from URL." Therefore, this reference can become `wp-image-caption-undefined`. Similar to guards elsewhere referenced below, should we guard against this case to only apply this value if there is an `id` set? 

https://github.com/WordPress/gutenberg/blob/30526b0e397f81b33ce349ef0086a98e8053ebb7/packages/block-library/src/image/save.js#L45-L47

If we add a guard here, we should likely it elsewhere for `describedById`, e.g. `packages/block-library/src/image/image.js`.